---
title: "Identifying composition rules for transcription factor circuits that control macrophage signal response with deep learning"
collection: publications
permalink: /publications/2018-07-31_genomic_grammar
venue: "International Workshop on Bio-Design Automation"
excerpt: 'This paper describes the application of a convolutional nerual network with an attention mechanism that can identify regulatory circuits in the genome with state-of-the-art performance. The attention mechanism of the neural network can also reveal the architecture of regulatory circuits within the genome'
date: 2018-07-31
paperurl: http://jenhantao.github.io/files/2018-07-31_genomic_grammar.pdf
---

## Abstract
Regulation of gene expression in mammalian cells is determined in part by the binding of combinations of transcription factors, which bind collaboratively to activate regulatory sequences called enhancers and recruit transcriptional machinery. Here we propose a deep learning method to uncover compositional rules for how the binding motifs of transcription factors are composed at these regulatory sequences. Using high throughput genomics assays, we will measure the response of the macrophage to various stimuli (cytokines), which form a critical component of the innate immune system. With this dataset characterizing the macrophage signal response, we aim to identify arrangements of transcription factor binding motifs derived from genomic sequence alone that can be used to predict the activation of macrophage regulatory elements specific to each signal. 

[Download paper here](http://jenhantao.github.io/files/2018-07-31_genomic_grammar.pdf)
